scripts/pre_commit_linter.py

```diff
@@ -2421,6 +2430,77 @@ def _check_extra_js_files(self):
 
         return summary_messages
 
+    def _check_constants_declaration(self):
+        """Checks the declaration of constants in the TS files to ensure that
+        the constants are not declared in files other than *.constants.ts.
+        """
+
+        if self.verbose_mode_enabled:
+            print 'Starting constants declaration check'
+            print '----------------------------------------'
+
+        summary_messages = []
+        failed = False
+
+        with _redirect_stdout(_TARGET_STDOUT):
+            all_ts_files = [
+                filepath for filepath in self.all_filepaths if (
+                    filepath.endswith('.ts'))]
+            ts_files_to_check = [
+                filepath for filepath in all_ts_files if not any(
+                    pattern in filepath for pattern in (
+                        OLD_CONVENTION_DIRECTORIES)) and filepath.endswith(
+                            'App.ts')]
```

Comment:
Not great to have this special case. Make it part of OLD_CONVENTION_DIRECTORIES?

Reply:
But then it is not a directory. It is not suited in the list. We would have to make a different list which would just have this file. Or change the name from OLD_CONVENTION_DIRECTORIES to OLD_CONVENTION_PATHS?

Reply:
Paths is fine.

Reply:
Done.